### PR TITLE
add (EPS/PEP/NEP)GetErrorEstimate

### PR DESCRIPTION
**Pull request by aszaboa**

### CODE REVIEW

File: plugin/mpi/SLEPc-code.hpp [hunk 1]
```diff
@@ -416,14 +420,24 @@ AnyType eigensolver<Type, K, SType>::E_eigensolver::operator()(Stack stack) cons
                 for(PetscInt i = 0; i < nconv; ++i) {
                     PetscScalar kr, ki = 0;
                     PetscReal sigma;
-                    if(std::is_same<SType, EPS>::value)
+                    PetscReal errest;
+                    if(std::is_same<SType, EPS>::value){
```
Comment (prj-) [inline plugin/mpi/SLEPc-code.hpp:424]:
```suggestion
                    if(std::is_same<SType, EPS>::value) {
```

File: plugin/mpi/SLEPc-code.hpp [hunk 2]
```diff
@@ -416,14 +420,24 @@ AnyType eigensolver<Type, K, SType>::E_eigensolver::operator()(Stack stack) cons
...
                         EPSGetEigenpair(eps, i, &kr, &ki, (eigenvectors || array) ? xr : NULL, (eigenvectors || array) && std::is_same<PetscScalar, double>::value && std::is_same<K, std::complex<double>>::value ? xi : NULL);
+                        if(errorestimate)
+                            EPSGetErrorEstimate(eps, i, &errest);
+                    }
                     else if(std::is_same<SType, SVD>::value)
                         SVDGetSingularTriplet(svd, i, &sigma, xr, xi);
-                    else if(std::is_same<SType, NEP>::value)
+                    else if(std::is_same<SType, NEP>::value){
```
Comment (prj-) [inline plugin/mpi/SLEPc-code.hpp:431]:
```suggestion
                    else if(std::is_same<SType, NEP>::value) {
```

File: plugin/mpi/SLEPc-code.hpp [hunk 3]
```diff
@@ -609,3 +625,4 @@ static void Init() {
      Init_PETSc();
 }
 #endif
+
```
Comment (prj-) [inline plugin/mpi/SLEPc-code.hpp:628]:
```suggestion
```
Unneeded blank line.

**Review by prj- — CHANGES_REQUESTED** [review]
Very nice, thank you for your contribution. See, C++ is not that hard after all :)
I've made some minor suggestions.

**Comment by aszaboa** [comment]
Great, I am glad the code is acceptable. Well, C++ is still extremely weird, but I hope to learn and contribute more.  
If the modified FreeFem example is also fine, then I think we can proceed.

**Pull request merged**